### PR TITLE
Red Oxygen Grenades

### PR DESCRIPTION
**Pull request by Azrun**
## About the PR 
Blue Red Oxygen Grenade

Adds an extremely fast means of providing oxygen tanks amount of oxygen to a space.  Required resources are intended to justify amount of resources provided.  Slapped balance on this due to allowing for mitigating efforts made by antagonists make an area uninhabitable/undesirable.

Distributes: 1500 Oxygen, 100 CO₂, at 35C° to target turf's airgroup (giving directly to group if group processing is enabled, or distributes to individual members if group processing is disabled). 

Causes misstep, ear damage, temporary deafness, and stamina loss relative to distance from explosion.

Item Incorporation:
* Added anecdote about using oxygen grenades to engineering pocket guide.
* Added engineering pocket guide to Engineers starting backpack.
* Added oxygen grenade to Engineers starting backpack.
* "4x Red Oxygen Grenades, x4 Metal Foam Grenades" Crate added to QM
* Scannable

Pending general green-light:
- [x] Add 3 oxygen grenades to technical storage/engineering or whatever engineering repair equipment is for in-rotation maps.

## Why's this needed? 
People complain frequently about the "difficulty" of pressurization of an area.  This complaint is well founded in the somewhat unintuitive nature of how the gas is usually spread in a space due to the current speed of our atmospheric simulation.  This acts as a solution to the problem without having to work though additional computational impact and balance of explosive decompression that would otherwise allow for gases to near immediately fill a volume.  Additionally avoids the fact that a gaseous flame would fully envelope the space burning until fuel sources are completed.

## Changelog

```changelog
(u)Azrun
(*)Added experimental red oxygen grenade to station repair equipment.  Violently releases large amounts of oxygen into an area.
(+)Red oxygen grenades are available from QM and each engineer starts with one.
```

### CODE REVIEW

**Comment by Kyle2143** [comment]
Yes, good idea. I've wanted to make this before, in fact, but never sat down and did it. Now I never have to, excellent!

**Comment by Flaborized** [comment]
I like the idea a lot, seems like it'd be useful enough to become a Thing Engineers Carry With Them Around so it might be good to equip them with it if there's any good slots for them to have one or two. (Would also make it more clear that fixing breaches is directly their job and ensure that engineers are fully aware of this useful tool). 

Visually I think having them stored in boxes will make them less easily identifiable. Less experienced players are much more likely to See the item if it's not sequestered in storage, so they'll be more likely to hover over it without picking it up and then understand that it's there, so they'll then (in theory) be able to think back to it being there when they would want to use it.

I also don't really understand it being Red, since basically everything relating to oxygen is marked blue. I get that you can't take the current sprite and make it blue because then it'd look exactly like a cryo-grenade, but it doesn't *have* to be a recolor of a cryo grenade. Could easily imagine it as a full-light-blue version of a sarin grenade or with it even having a unique shape, to better fit the visual language we've established for air/oxy related items

**Comment by Azrun** [comment]
Super agree with the identifiable being lost in the box of grenades.  I was originally going to sprinkle them around but thought it would be too busy and decided to repackaging them would be preferable.  Surfacing them visually is much better, it helped me identify and focus on the metal foam grenades for instance.

Under EXTREME pressure the gas changes from a blue to red. https://en.wikipedia.org/wiki/Octaoxygen From a keying off of science I think it is interesting, but I do get what you are saying as a departure of typical coloration and iconography for oxygen.  Thought was an interesting nod to science.  I liked the cryo-grenade as it also allowed you to see the contents of the "chamber" and a reaction take place.

**Comment by ThatFiveGuys** [comment]
This should probably be tossed into a cargo crate, perhaps have 1 in atmospherics supplies?

File: code/WorkInProgress/explosion_particles.dm [hunk 1]
```diff
@@ -56,6 +56,16 @@
...
+	custom
+		New(turf/newLoc, color=null, plane=null, scale=null)
+			..()
+			if(color)
+				src.color = color
+			if(plane)
+				src.plane = plane
+			if(scale)
+				src.transform = matrix(scale, MATRIX_SCALE)
+
```
Comment (Azrun) [inline code/WorkInProgress/explosion_particles.dm:68]: These aren't necessary as caller can overwrite this directly.

File: code/obj/item/book.dm [hunk 2]
```diff
@@ -3361,7 +3361,7 @@ On that note, CentComm would like to remind trainees that they do not recieve cl
 		Damaged windows may be repaired by applying glass to a grille; if the grille is damaged or missing, repair it with metal rods from sheets or storage.
 		<br>
 		<br>
-		Once all breaches have been repaired, obtain and utilize an air canister (white, potentially with a red stripe) and space HVAC on heat setting.
+		Once all breaches have been repaired, obtain and utilize an air canister (white, potentially with a red stripe) and space HVAC on heat setting.  Alternatively utilize an oxygen grenade ensuring no associates are in blast zone prior to activation.
```
Comment (Azrun) [inline code/obj/item/book.dm:3364]:
```suggestion
		Once all breaches have been repaired, obtain and utilize an air canister (white, potentially with a red stripe) and space HVAC on heat setting.  Alternatively utilize an oxygen grenade ensuring no staff members are in the blast zone prior to activation.
```
@nefarious6th thoughts?

File: code/obj/item/grenades.dm [hunk 3]
```diff
@@ -550,12 +550,76 @@ PIPE BOMBS + CONSTRUCTION
...
+		GM.temperature = T20C + 15
+		GM.oxygen = 1500
+		GM.carbon_dioxide = 100
+
+		// Adjust coloration of explosion to lighter cyan
+		var/color_matrix = list(0,0,0.6,0.2,0, 0.3,0,-0.8,0,0, 0,0.33,0,0,0, 0.9,0.8,0.8,0.7,0)
+		var/obj/effects/explosion/E = new /obj/effects/explosion/custom(T,plane=PLANE_NOSHADOW_ABOVE,scale=0.75,color=color_matrix)
+		animate_wave(E,4)
+
+		if (T && istype(T))
```
Comment (Azrun) [inline code/obj/item/grenades.dm:577]: Use consistent style

File: code/obj/item/book.dm (hunk 2 above)
Comment (nefarious6th) [reply code/obj/item/book.dm:3364]: Up to you! Missed this earlier, I think "associates" has the right detached and non-caringness that might come with an outside firm (Servin Underwriting) preparing the instructional documents. That gets into the weeds on it though, I think either works fine.

**Comment by Yellow-Mushroom** [comment]
>Red oxygen grenade

>video is of a blue oxygen grenade 
🤔

but in all seriousness, is there any way to produce more of these?

**Comment by Azrun** [comment]
> but in all seriousness, is there any way to produce more of these?

Scannable, QM, and... plan to add handful to maps.

**Comment by Tarmunora** [comment]
Hm. If the group isn't in group processing, would it be reasonable to limit it to only repressurizing tiles within a certain radius (range 7?), or would that defeat the purpose too much

**Comment by Azrun** [comment]
Yes and yes.  
Limiting to a range of 7 would make it behave in a manner more in line with respect to the current systems.  It might also ensure the area is immediately breathable...

One of the goals was to try to get around the current behavior (https://discord.com/channels/182249960895545344/298827721604071425/826899523489497118) and allow for immediate dispersion in an area.  A range of 7 pretty much achieves that for probably +90% of rooms.  Notable exceptions would be places like the chapel, medbay, and hallways where depending on where you put the grenade you would need to wait for atmos to merge the gasses out which was part of the current frustration.  If additional grenade(s) are needed the player will know very quickly.

I'll chew on it and am definitely open to opinions.

**Comment by virvatuli-gh** [comment]
As someone who frequently complains about the state of our atmos system and how anti-fun repressurizing is, I think it's a really stellar feature that will get used *a lot*. I also like that they actually act like grenades with downside to standing next to one when it goes off.

**Pull request merged**